src/Core/Settings.h

```diff
@@ -260,6 +260,8 @@ class IColumn;
     M(Bool, force_primary_key, false, "Throw an exception if there is primary key in a table, and it is not used.", 0) \
     M(Bool, use_skip_indexes, true, "Use data skipping indexes during query execution.", 0) \
     M(Bool, use_skip_indexes_if_final, false, "If query has FINAL, then skipping data based on indexes may produce incorrect result, hence disabled by default.", 0) \
+    M(Bool, materialize_skip_indexes_on_insert, true, "If true skip indexes are calculated on inserts, otherwise skip indexes will be calculated only during merges", 0) \
```

Comment:
Maybe it make sense to accept list of indexes to skip? since most of the indexes are light-weight, unlike i.e. annoy/usearch), and likely users will want to skip them, and indeed it make a big sense to write them only during merges I guess